lib/include/openamp/rpmsg.h

```diff
@@ -161,8 +161,11 @@ int rpmsg_send_offchannel_raw(struct rpmsg_endpoint *ept, uint32_t src,
 static inline int rpmsg_send(struct rpmsg_endpoint *ept, const void *data,
 			     int len)
 {
-	return rpmsg_send_offchannel_raw(ept, ept->addr, ept->dest_addr, data,
-					 len, true);
+	if (ept)
+		return rpmsg_send_offchannel_raw(ept, ept->addr, ept->dest_addr, data,
+						 len, true);
+	else
+		return RPMSG_ERR_PARAM;
```

Comment:
suggestion:
```
if (!ept)
	return RPMSG_ERR_PARAM;

return rpmsg_send_offchannel_raw(ept, ept->addr, ept->dest_addr, data,
						 len, true);

```
comment to apply for the other function below. when possible test input parameters first and return on error

Reply:
I am happy to change this per your request, but I generally test the most likely condition first in order to better optimize the code.  In your example, since it is more likely that ept will not be NULL, I am testing the most likely condition first.  Please let me know your thoughts.

Reply:
I''m wonder if something change in term of code optimization as it is a simple condition in which you have to test the ept for both case 
Do you check the delta?
that said defining "likely" & "unlikely" macro as done in Linux cool be something interesting for this kind of check.